### ISSUE TITLE
IBM PC, Original release of Nobunaga no Yabou: Ranseiki possibly filling non-erroneous sectors with 0x55, producing non-matching dumps

### ISSUE THREAD

**Issue by HeroponRikiBestest** [superg/redumper#164]
I apologize in advance for this post likely being confusing to read, as I'm not sure the best way to get everything across.

TL;DR, I have a disc that, while matching across many dumps, will still have many non-matching dumps. Each dump has 0 C2 errors, and (except for the very first dump I attempted) no subcode desyncs, either, nor did they need to be refined. Each dump appears to have some sections that are 0x55'd that the other dumps don't, and vice versa. If I split all of the dumps with --leave-unchanged, they will all match each other, regardless of whether they were matching before. The disc is protected with Safedisc (SafeDisc 1.50.020), but I have no idea if this is relevant at all.

In more detail:

I'm attempting to dump the original, non-sourcenext version of http://redump.org/disc/89816/. My disc has the same safedisc version as well in case it's relevant, SafeDisc 1.50.020. I dumped it 2 times on my ASUS, and 3 times on my 4824. It never produced errors on any of them, but it kept producing different dumps; and when i compare them against each other in a hex editor, every dump has a section that is errored out that isn't in one of the other dumps. It is protected, but with safedisc, which redumper correctly doesn't refine, and none of the non matching sections are anywhere near it anyways. I did get the same hash between both asus dumps and my third 4824 dump, but as mentioned, it has errored out sections that aren't errored out in the other 2 plextor dumps. The disc certainly isn't pristine and has some scratching, but nothing that visibly seems like it would guarantee errors; either way, I'd assume that the errors should they happen at least manifest in the form of c2 errors or subcode desyncs. There was one small subcode desync on the first asus dump, but not on any of the others. 

[nobunaga.log](https://github.com/user-attachments/files/15710756/nobunaga.log)
[nobunagaASUS.zip](https://github.com/user-attachments/files/15710757/nobunagaASUS.zip)

[nobunaga.log](https://github.com/user-attachments/files/15710758/nobunaga.log)
[nobunagaASUS2.zip](https://github.com/user-attachments/files/15710759/nobunagaASUS2.zip)

[nobunaga.log](https://github.com/user-attachments/files/15710764/nobunaga.log)
[nobunagaPLEX.zip](https://github.com/user-attachments/files/15710763/nobunagaPLEX.zip)

[nobunaga.log](https://github.com/user-attachments/files/15710767/nobunaga.log)
[nobunagaPLEX2.zip](https://github.com/user-attachments/files/15710768/nobunagaPLEX2.zip)

[nobunaga.log](https://github.com/user-attachments/files/15710770/nobunaga.log)
[nobunagaPLEX3.zip](https://github.com/user-attachments/files/15710772/nobunagaPLEX3.zip)

What i mean about the non matching sections being in all dumps:
(in both screenshots, ASUS/ASUS2/PLEX3 is on the left, crc32 ff5eb198, PLEX2 on the right, crc32 24933b5b) 
![image](https://github.com/superg/redumper/assets/50224630/c747d1c6-6383-495c-9e11-47c4bf0f534d)
![image](https://github.com/superg/redumper/assets/50224630/74fbb2bb-2994-45eb-bb14-0f40bf315c67)

I tried the latest release of DiC as well, and it got the same hash as ASUS/ASUS2/PLEX3 on both my asus and my 4824.

[DiC_nobunagaASUS.zip](https://github.com/user-attachments/files/15710801/DiC_nobunagaASUS.zip)

[DiC_nobunagaPLEX.zip](https://github.com/user-attachments/files/15710802/DiC_nobunagaPLEX.zip)

My redumper dumps were done using redumper v2024.05.06 build_325 [May  6 2024, 01:13:23], so I dumped it one more time on the latest release, redumper v2024.05.27 build_371 [May 27 2024, 14:06:13], to confirm the issue was still present. It is, and it produced a unique hash to all my other existing dumps, which likewise also has some sections in it that are 0x55 that aren't in the others and has some sections in it that aren't 0x55'd that are in others.

[nobunaga.log](https://github.com/user-attachments/files/15710826/nobunaga.log)
[nobunaga_371_PLEX.zip](https://github.com/user-attachments/files/15710824/nobunaga_371_PLEX.zip)

However, if i split my dumps again with --leave-unchanged, they all match each other, resulting in a matching hash, no matter what the original hash was when dumping. Unfortunately, I can't submit this to redump or provide a final hash, as I don't know of any way to split specifically only the part past the safedisc sectors, since the safedisc sectors should obviously be left as 0x55.

Please let me know what additional information needs to be clarified, what additional information/files need to be provided, and what additional things I need to try. I apologize again for this issue likely containing excessive or superfluous information.

**Comment by HeroponRikiBestest** [comment]
Not entirely sure what was going on with this, but it seems like the dump i was able to get in the end was probably good.

**Comment by superg** [comment]
As you probably know, SafeDisc protection intentionally corrupts some portion of tha disc with C2 errors.
Currently redump standard is to dump those areas precisely e.g. try to get as much good sectors at the start and the end of the erroneous area. This can be done only by continuous reading of all C2 sectors. Drives do not like C2 errors and depending on a drive, drive speed and current position of the stars you might get a deviation, usually +- 1 sector. This method doesn't give 100% result every time but we declare it "good enough". Same thing happens with DIC sometimes, I don't know precisely how DIC is reading, if it seeks back and forward it might behave a little bit differently but overall it's also not immune to this issue. What I mean is that this problem is not specific to redumper.

More correct way of handling it is to extend those areas to the sectors that we know are unused on the disc, surrounding sectors are zeroed anyway (but scrambled) but such an approach will require changing all current SafeDisc submissions. It can be done programatically for those discs that have roms publically available but somebody has to take a lead on that.

> However, if i split my dumps again with --leave-unchanged, they all match each other
This is probably a coincidence and will not be a rule for the general case.

**Comment by HeroponRikiBestest** [comment]
The affected areas were nowhere close to the safedisc error range; everything in and near that range was handled fine every time.

**Comment by superg** [comment]
So you got a dummy sector elsewhere on a disc?
Don't have much time to look into your files.

**Comment by HeroponRikiBestest** [comment]
Probably don't worry about it. It's more likely I just messed something up somehow; I closed the issue because I don't think it's worth looking into, unless you feel otherwise.
